internal/schedule/penalize/penalize.go

```diff
@@ -0,0 +1,39 @@
+package penalize
+
+import (
+	"github.com/NodeFactoryIo/vedran/internal/active"
+	"github.com/NodeFactoryIo/vedran/internal/models"
+	"github.com/NodeFactoryIo/vedran/internal/repositories"
+	log "github.com/sirupsen/logrus"
+	"time"
+)
+
+const MaxCooldownForPenalizedNode = 17 * time.Hour
+
+var afterFunc = time.AfterFunc
+
+func ScheduleCheckForPenalizedNode(node models.Node, repositories repositories.Repos) {
+	afterFunc(time.Duration(node.Cooldown), func() {
+		isActive, err := active.CheckIfNodeActive(node, &repositories)
+		if err != nil {
+			log.Errorf("Unable to check if node %s active, because of %v", node.ID, err)
+		}
+
+		if isActive {
+			repositories.NodeRepo.AddNodeToActive(node)
+		} else {
+			newCooldown := node.Cooldown * 2
+			if (time.Duration(newCooldown) * time.Minute) > MaxCooldownForPenalizedNode {
+				log.Debugf("Node %s reached maximum cooldown", node.ID)
+				// TODO - remove node
```

Comment:
Can you add this?

Reply:
I think we should do this in a separate issue, as whitelisting mechanics need to be refactored so here we can remove the node from the whitelist. I will make issues for that.